src/display/fetch_stream.js

```diff
@@ -248,10 +248,7 @@ class PDFFetchStreamRangeReader {
         this._reader = response.body.getReader();
       })
       .catch(reason => {
-        if (reason?.name === "AbortError") {
```

Comment:
This code was added in PR #12136, to get rid of one source of "Uncaught (in promise) DOMException: ..." errors from being logged while the unit-tests are run.

Assuming that this particular change first of all doesn't break any existing tests, and secondly that those errors messages don't start showing up again this is probably fine.
However, I'd like confirmation that none of the ways that the unit-tests can be run now triggers the "Uncaught (in promise) DOMException: ..." errors; which include:
 - Running `gulp unittest`, and checking the browser consoles while the tests run.
 - Execute `gulp server` such that the unit-tests can be run manually via http://localhost:8888/test/unit/unit_test.html, and checking the browser consoles while the tests run.
 - Run `gulp unittestcli`, and checking that there's no Promise-warnings.

---

As for the complete patch, I'm still thinking about the overall approach since this `src/core/chunked_stream.js` file contains code that we really do *not* want to accidentally break. (It'd be nice if the patch included unit-tests for the things it fixes, although I'm not sure how feasible that'd be to implement unfortunately.)

Reply:
> Assuming that this particular change first of all doesn't break any existing tests, and secondly that those error messages don't start showing up again this is probably fine.

As mentioned in the OP, all tests (but a few pixel-by-pixel comparisons) pass. I did not spot any new error messages.

> However, I'd like confirmation that none of the ways that the unit-tests can be run now triggers the "Uncaught (in promise) DOMException: ..." errors; which include:
> 
> * Running `gulp unittest`, and checking the browser consoles while the tests run.
> * Executing `gulp server` such that the unit-tests can be run manually via http://localhost:8888/test/unit/unit_test.html, and checking the browser consoles while the tests run.
> * Running `gulp unittestcli`, and checking that there's no Promise-warnings.

I ran all of the above with success (except two failures in the manual run related to different time zones [1], they are failing on the `master` branch as well). No uncaught exceptions were logged. 

<details>
<summary> browser console </summary>

![image](https://user-images.githubusercontent.com/17931887/131323191-efde6ea2-a7cf-4c7a-b3f4-97dc7f6f32c5.png)
</details>


> As for the complete patch, I'm still thinking about the overall approach since the `src/core/chunked_stream.js` file contains code that we really do _not_ want to accidentally break. (It'd be nice if the patch included unit-tests for the things it fixes, although I'm not sure how feasible that'd be to implement unfortunately.)

I can have a look at adding tests for the changed error handling.

---

[1] 
<details>

![image](https://user-images.githubusercontent.com/17931887/131322996-77de6c13-7498-415b-a2c8-b037340164fc.png)
</details>

Reply:
> I can have a look at adding tests for the changed error handling.

I don't think we can directly test either of the two cases specifically mentioned in https://github.com/mozilla/pdf.js/pull/13937#issue-720538682 without quite a lot of other changes to the test-suite, hence we could probably defer that to a follow-up instead (otherwise there's a risk that landing this patch could be delayed a fair bit).

---

> We are seeing good results with the patch (rebased on top of v2.2.288) in production over the course of the past 2 months.

Slightly unrelated, but note that the mentioned version is *over two years old* and really shouldn't be used any more.

Reply:
I've added some tests for the changed handling in `fetch_stream.js` via 60c1bbe684e46ef7ae1d72f12bf7cd95bbd39455.
These tests are failing on master.

![image](https://user-images.githubusercontent.com/17931887/131330033-76958138-6a09-4c9c-a3de-df827bb80760.png)
![image](https://user-images.githubusercontent.com/17931887/131330073-c3a52ae6-7bc2-4a60-9dd1-6ad6c53eada9.png)

Reply:
> I've added some tests for the changed handling in `fetch_stream.js` via [60c1bbe](https://github.com/mozilla/pdf.js/commit/60c1bbe684e46ef7ae1d72f12bf7cd95bbd39455).
> These tests are failing on master.

Thanks, however I'm unfortunately not sure that they actually help all that much in this case since what we *really* wanted to test here is the `src/core/chunked_stream.js` changes (which as mentioned we currently don't have a way to do).

I'm also not crazy about overriding the native Fetch API in this way; all-in-all I'd actually suggest we skip tests for now!

Reply:
OK, I've force-pushed them away now.